xbmc/epg/GUIEPGGridContainer.cpp

```diff
@@ -1896,6 +1896,12 @@ void CGUIEPGGridContainer::Reset()
 
   m_lastItem    = NULL;
   m_lastChannel = NULL;
+
+  m_channels = 0;
+  m_channelCursor = 0;
+  m_channelOffset = 0;
+  m_blockCursor = 0;
+  m_blockOffset = 0;
```

Comment:
This kills the logic which restores the epg event selection after grid data recreation => https://github.com/xbmc/xbmc/blob/master/xbmc/epg/GUIEPGGridContainer.cpp#L1025 ff as this code (for instance <code>GetSelectedEpgInfoTag</code> )  currently relies on m_channelCursor, m_channelOffset, m_blockCursor, m_blockOffset not being reset before grid recreation. Solution could be to remember these values locally before calling Reset()?